presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,26 +13,80 @@
  */
 package com.teradata.prestomanager.agent;
 
+import com.google.common.io.ByteStreams;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 
-import static java.lang.Runtime.getRuntime;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
-// TODO: Add logger
 public final class CommandExecutor
 {
-    private CommandExecutor() {}
+    private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+    private final String[] commandArray;
+    private final int timeout;
 
-    public static int executeCommand(String command)
+    private CommandExecutor(String[] commandArray, int timeout)
+    {
+        this.commandArray = requireNonNull(commandArray);
+        this.timeout = timeout;
+    }
+
+    static int commandExecutor(String... command)
             throws PrestoManagerException
     {
+        return new CommandExecutor(toCommandArray(command), 60).execute();
+    }
+
+    static int commandExecutor(int timeoutInSeconds, String... command)
+            throws PrestoManagerException
+    {
+        return new CommandExecutor(toCommandArray(command), timeoutInSeconds).execute();
+    }
+
+    private static String[] toCommandArray(String... command)
```

Comment:
What is this function's purpose? Does it do anything besides turn an array into a list and back again?